src/TuxxState.js

```diff
@@ -0,0 +1,340 @@
+'use strict';
+
+var assign = require('object-assign');
+var invariant = require('tuxx/src/TuxxInvariant');
+
+// buildNewState FUNCTION: internal function that creates an updated state object by invoking callback on the deepest keys of the newProps object, throwing an error if any of the keys in newProps dont match the corresponding keys in currentState
+// @param currentState OBJECT: a shallow copy of this.state
+// @param newProps OBJECT: an object holding keys that should match those in currentState and whose values will each be passed to callback
+// @param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+var buildNewState = function (currentState, newProps, callback) {
+  // create empty object which will be used to build the eventual state object that will be passed in to setState after our callback has been invoked on the proper deep keys
+  // NOTE: React's setState method makes a shallow copy of the keys at the top-most level, so we don't need to make a shallow copy of the top-most keys here, but rather, make shallow copies of keys at every level EXCEPT for the top
+  var newState = {};
+
+  // recurseKeys FUNCTION: traverses through keys in a heavily nested object, building our newState object as we iterate through potential nested objects and setting the deepest keys passed by newProps to be the result of invoking our callback on the deepest keys in the newProps and corresponding currentState keys
+  // @param currentST OBJECT: properties of the currentState
+  // @param newST OBJECT: properties of the newState
```

Comment:
Change newSt to newState either that or nextState if it is the next state of componet.